record/record.go

```diff
@@ -117,6 +118,9 @@ type Reader struct {
 	n int
 	// started is whether Next has been called at all.
 	started bool
+	// inRecovery is true when Recover() has been called, but before
```

Comment:
Drop the (). In Go comments, unlike C++ comments, "foo" usually refers to the method, and "foo()" refers to the result of calling that method with no arguments.

Ditto below.